cmd/brew-report-issue.rb

```diff
@@ -13,19 +13,21 @@
   body = STDIN.read
 end
 
-@credential_helper = `git config --global credential.helper`.chomp
-
 strap_url = ENV["STRAP_URL"]
 strap_url ||= "https://strap.githubapp.com"
 
-if @credential_helper.empty?
+if `git config --global credential.helper`.chomp.empty?
   abort <<-EOS
 Error: your Git HTTP(S) credential helper is not set! Set it by running Strap:
 #{strap_url}
 EOS
 end
 
-@github_credentials=`printf "protocol=https\nhost=github.com\n" | git credential fill 2>/dev/null`
+def credential_helper(command, input)
+  `printf "#{input}" | git credential #{command} 2>/dev/null`
```

Comment:
In the original, the shell `printf` was converting `\n` to newline, but you can do that in ruby, and get rid of `printf` entirely. And that saves you the interpolation headaches. E.g.:

```ruby
# will mess up some strings
def one(cmd, input)
	`printf "#{input}" | #{cmd}`
end

# will feed input verbatim to command
def two(cmd, input)
	IO.popen(cmd, mode="w+") {|io|
		io.print input
		io.close_write
		io.read
	}
end

# just a stand-in for git-credential
cmd = 'sed "s/^/sed: /"'
# this has some tricky sequences in it; the single
# quote means ruby doesn't interpolate \n
msg = 'something with %s and \n in it'

print "one: ", one(cmd, msg), "\n"
print "two: ", two(cmd, msg), "\n"

# but here we use double quotes, and ruby expands \n
print two(cmd, "ruby will interpolate this\nnewline; no need for printf")
```

which should produce:

```
one: sed: something with  and 
sed:  in it
two: sed: something with %s and \n in it
sed: ruby will interpolate this
sed: newline; no need for printf
```

Does that make sense?

I'm not much of a ruby programmer, so my use of popen and general style may not be idiomatic.